flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/AccessExecution.java

```diff
@@ -64,9 +66,11 @@
      * Returns the exception that caused the job to fail. This is the first root exception that was
      * not recoverable and triggered job failure.
      *
-     * @return failure exception as a string, or {@code "(null)"}
+     * @return an {@code Optional} of {@link ErrorInfo} containing the {@code Throwable} wrapped in
+     *     a {@link SerializedThrowable} and the time it was registered if an error occurred. If no
```

Comment:
nit: That the throwable is wrapped in a `SerializedThrowable` should be an implementation detail of the `ErrorInfo`. I think it is not necessary to mention it here.